tests/python/test_fields_builder.py

```diff
@@ -3,174 +3,259 @@
+
+                    for create_field_idx in range(field_size):
+                        field = ti.field(field_type)
+                        fb.dense(ti.i, test_1d_size).place(field)
+                    fb_snode_tree = fb.finalize()
+        fb_snode_tree.destroy()
+
+    test_for_single_destroy_multi_fields(test_sizes, size_extend_factor, field_types, field_sizes)
+
+    # [destroy] 2. test for multiple destroy for multiple fields
+    def test_for_multi_destroy_multi_fields(test_sizes, size_extend_factor, field_types, field_sizes): #size_1d_0, size_1d_1):
+        fb0 = ti.FieldsBuilder()
+        fb1 = ti.FieldsBuilder()
+
+        for tsize_idx in range(len(test_sizes)):
+            for ftype_idx in range(len(field_types)):
+                for fsize_idx in range(len(field_sizes)):
+                    test_1d_size = test_sizes[tsize_idx] * size_extend_factor
+                    field_type = field_types[ftype_idx]
+                    field_size = field_sizes[fsize_idx]
+
+                    for create_field_idx in range(field_size):
+                        field0 = ti.field(field_type)
+                        field1 = ti.field(field_type)
+
+                        fb0.dense(ti.i, test_1d_size).place(field0)
+                        fb1.pointer(ti.i, test_1d_size).place(field1)
+
+                        fb0_snode_tree = fb0.finalize()
+                        fb1_snode_tree = fb1.finalize()
+
+        # destroy
+        fb0_snode_tree.destroy()
+        fb1_snode_tree.destroy()
+
+    test_for_multi_destroy_multi_fields(size_1d_0, size_1d_1)
+
+    # [destroy] 3. test for raising second destroy
+    def test_for_raise_twice_destroy(size_1d):
+        fb = ti.FieldsBuilder()
+        a = ti.field(ti.f32)
+        fb.dense(ti.i, size_1d).place(a)
         c = fb.finalize()
+        id0 = c.id()
         c.destroy()
 
-    def B(i):
-        n = i * 10**3
-        fb = ti.FieldsBuilder()
-        a = ti.field(ti.f64)
-        fb.dense(ti.i, n).place(a)
-        c = fb.finalize()
+        with pytest.raises(InvalidOperationError) as e:
+            id_1 = c.id()
+            c.destroy()
 
-        ni = i * 10**3
-        fbi = ti.FieldsBuilder()
-        ai = ti.field(ti.f64)
-        fbi.dense(ti.i, n).place(ai)
-        ci = fbi.finalize()
+    test_for_raise_twice_destroy(10)
 
-        c.destroy()
-        ci.destroy()
 
-    for i in range(5):
-        A(5)
-    B(2)
-    A(4)
+'''
+Test fields with builder exceeds max.
+'''
+@ti.test(arch=[ti.cpu, ti.cuda])
+def test_fields_builder_exceeds_max():
+    shape_size = (4, 4)
+
+    def create_fb():
+        fb = ti.FieldsBuilder()
+        x = ti.field(ti.f32)
+        fb.dense(ti.ij, shape_size).place(x)
+        fb.finalize()
+
+    # kMaxNumSnodeTreesLlvm=32 in taichi/inc/constants.h
+    # TODO(ysh329): kMaxNumSnodeTreesLlvm=512 not 32 in `taichi/inc/constants.h`.
+    #               Can this value `kMaxNumSnodeTreesLlvm` load from `.h` file?
```

Comment:
kMaxNumSnodeTreesLlvm=512 not 32 in `taichi/inc/constants.h`.
               Can this value `kMaxNumSnodeTreesLlvm` load from `.h` file?

Reply:
This test has been removed due to long execution time :-)

Reply:
OKay